stl/inc/iterator

```diff
@@ -1026,15 +1036,24 @@ struct _Common_iterator_pointer_type<_Iter> {
     using pointer = decltype(_STD declval<_Iter&>().operator->());
 };
 
-template <input_iterator _Iter, class _Se>
-struct iterator_traits<common_iterator<_Iter, _Se>> {
-    using iterator_concept = conditional_t<forward_iterator<_Iter>, forward_iterator_tag, input_iterator_tag>;
+template <class _Iter>
+struct _Common_iterator_traits_category_base {
+    using iterator_category = input_iterator_tag;
+};
+
+template <_Has_member_iterator_category _Iter>

```

Comment:
`_Iter::iterator_category` doesn't necessarily need to exist for `iterator_traits<_Iter>::iterator_category` to exist. (This is a P2259 issue really.)

Reply:
I have taken over the resolution of P2259.